app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -11,19 +11,129 @@ import 'package:github/github.dart';
 import '../proto/internal/scheduler.pb.dart' as pb;
 import 'target.dart';
 
+/// Key used for selecting which .ci.yaml to use
+enum CiType {
+  /// "Default" associated with this slug / ci.yaml
+  ///
+  /// Pre-fusion: engine, framework, etc
+  /// Post-fusion: framework, etc
+  ///
+  /// NOTE: Required to be present
+  any,
+
+  /// Engine's .ci.yaml file located in the monorepo
+  fusionEngine,
+}
+
+/// Wrapper arround one or more [CiYaml]
```

Comment:
Can you clarify more? Why do we need an abstraction around multiple CiYamls?

Also, should maybe CiYamlInner be named CiYaml, and this be named something else, maybe CiYamlSet?

Reply:
> Can you clarify more

Yes, I'll take a stab

> CiYaml vs CiYamlSet

I originally wanted a smaller diff-set and file changes.  If I change it to CiYamlSet, would folks expect it to actually be a Set<CiYaml>?  I'll give it a stab and see how it looks (at the call site, I'm sure it'll read better).